libs/git/info.go

```diff
@@ -0,0 +1,124 @@
+package git
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"net/http"
+	"strings"
+
+	"github.com/databricks/cli/libs/dbr"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/cli/libs/vfs"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/client"
+)
+
+type GitRepositoryInfo struct {
+	OriginURL     string
+	LatestCommit  string
+	CurrentBranch string
+	WorktreeRoot  vfs.Path
+}
+
+type gitInfo struct {
+	Branch       string `json:"branch"`
+	HeadCommitID string `json:"head_commit_id"`
+	Path         string `json:"path"`
+	URL          string `json:"url"`
+}
+
+type response struct {
+	GitInfo *gitInfo `json:"git_info,omitempty"`
+}
+
+func FetchRepositoryInfo(ctx context.Context, path vfs.Path, w *databricks.WorkspaceClient) (GitRepositoryInfo, error) {
+	if strings.HasPrefix(path.Native(), "/Workspace/") && dbr.RunsOnRuntime(ctx) {
+		return FetchRepositoryInfoAPI(ctx, path, w)
+	} else {
+		return FetchRepositoryInfoDotGit(ctx, path)
+	}
+}
+
+func FetchRepositoryInfoAPI(ctx context.Context, path vfs.Path, w *databricks.WorkspaceClient) (GitRepositoryInfo, error) {
```

Comment:
I find it easier & safer to deal with full structs than with pointers (no possibility of crash due to to pointer being nil). Performance-wise it should also be better since you don’t trigger heap operations and GC. Some additional discussion [here](https://forum.golangbridge.org/t/returning-value-or-pointer-struct-for-crud-methods/35194/3) and [here](https://medium.com/@philpearl/bad-go-pointer-returns-340f2da8289).